### ISSUE TITLE
Generator type bug

### ISSUE THREAD

**Issue by mhzed** [Microsoft/pyright#101]
**Describe the bug**
The return type of a generator function is "Iterator[...]" type without member function 'close'.

**To Reproduce**
```python
def generate():
  for i in range(2):
    yield i

s = generate()
s.close()   # valid python code,  but pyright reports "Cannot access member close on type 'Iterator[int]'
```
**Expected behavior**
No error reported.

**Screenshots or Code**

**VS Code extension or command-line**
vscode 

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. Currently, pyright's type inference logic assumes that a generator function returns an Iterator[Y] where Y is the type returned by the yield statement(s). More correctly, it should infer the type Generator[R, Y,] where R is the return type and Y is the yield type. 

As a temporary workaround, you can annotate your generator's return function with "Generator[None, int]", and the error will no longer be reported.

**Comment by erictraut** [comment]
This is fixed in version 1.0.21, which I just published.
